### ISSUE TITLE
Crash On Segue to CollectionView

### ISSUE THREAD

**Issue by Crossroadsman** [Crossroadsman/ios10AndSwift3FromBeginnerToPaidProfessional#5]
# Issue: App crashes when loading collection view #

## Date Seen ##
2016-10-07 11:45 MST

## Versions ##
*observed on*
XCode 8.0 (build 8A218a)
MacOS 10.12.0 (build 16A323)
Mac Pro (late 2013)
Git commit 972b673049a20325e4f6e7667cc21226de25bea6

*not observed on*
No other devices tested

## Issue Description ##
App crashes when the pokeball button is pressed (which triggers segue to collection view via navigation controller)

## Severity ##
Fatal

## Steps to Reproduce ##
1. Run App
2. Press Pokeball button

## Actual Behaviour ##
Crash (see crash report below)

## Expected Behaviour ##
Collection view embedded in navigation controller should load

## Testing Steps Attempted ##
- TODO: Try building basic app with just the specific connections in this app, see if crash occurs
**Update 2016-10-07 15:10 MST:**
Tried this: same result.

- TODO: Try adding FlowLayoutDelegate functionality
**Update 2016-10-07 15:10 MST:**
Tried this: same result.

**Update 2016-10-07 15:10 MST:**
After trying several mini apps to isolate the problem (see commit: 7c6c96fdb5b266e4eafd448d85bc1bb84fffc228), it seems possible that the crash is caused by the CollectionView controller being specified as of type UICollectionViewController. Creating an app where the collectionview is specified as UIViewController and the required CollectionView methods are implemented did not cause a crash.

- TODO: Change type of viewcontroller in real app to UIViewController
**Update 2016-10-07 15:18 MST:**
Done (commit: 1cfd4860c11e1cf9ffd3880a28fc04d4c966b7a8). This fixed the bug.

- TODO: Try replacing programmatic segue with storyboard segue

## Workaround ##
None identified

### Xcode Crash Data ###
```
** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: '-[UICollectionViewController loadView] loaded the "Zcw-OA-TMZ-view-h8A-bX-yMq" nib but didn't get a UICollectionView.'
```

**Comment by Crossroadsman** [comment]
Possible fix found. See update comments. Now flowing possible success into main app.

**Comment by Crossroadsman** [comment]
Fixed in commit 1cfd4860c11e1cf9ffd3880a28fc04d4c966b7a8! (App still crashes, but now due to a pedestrian index out-of-bounds error)
